token/cli/src/main.rs

```diff
@@ -7425,4 +7453,64 @@ mod tests {
                 .unwrap()
         );
     }
+
+    #[tokio::test]
+    #[serial]
+    async fn metadata_pointer() {
+        let (test_validator, payer) = new_validator_for_test().await;
+        let config =
+            test_config_with_default_signer(&test_validator, &payer, &spl_token_2022::id());
+        let program_id = &spl_token_2022::id();
+        let token_keypair = Keypair::new();
+        let token_path = NamedTempFile::new().unwrap();
+        let token_pubkey = token_keypair.pubkey();
+        let bulk_signers: Vec<Arc<dyn Signer>> =
+            vec![Arc::new(clone_keypair(&payer)), Arc::new(token_keypair)];
+        let metadata_address = Pubkey::new_unique();
+
+        command_create_token(
+            &config,
+            TEST_DECIMALS,
+            token_pubkey,
+            payer.pubkey(),
+            false,
+            true,
+            false,
+            false,
+            None,
+            Some(metadata_address),
+            None,
+            None,
+            None,
+            None,
+            bulk_signers,
+        )
+        .await
+        .unwrap();
```

Comment:
Rather than using `command_create_token`, which doesn't exercise the command line parsing, can you execute using `process_test_command`, ie:

```
process_test_command(
    &config, &payer,
    &["spl-token", CommandName::CreateToken.into(), "--metadata-address", metadata_address.to_string()]
).await;
```

You're doing this later in the test, so can you remove this and move that call up here? You'll just need to extract the mint info, just like in the `create_token_default` test

Reply:
Is it okay to omit the `--program_id` arguments?

Reply:
Ah, no, definitely not